doc/readDaq.md

```diff
@@ -0,0 +1,151 @@
+- `data` outputs the data that was read as an array of elements.
+The user will allocate the memory by himself/herself in this method.
+- `numberOfElements` is the number of elements to read
+- `timeout` specifies the time in milliseconds the function waits for available data in the FIFO, if the FIFO is empty
+- `elementsRemaining` outputs the number of elements remaining in the host memory part of FIFO
+
+Note:
+
+- Session has to be oppened with:
+
+```c
+NiFpga_Session session;
+```
+
+- Session must be closed when no longer needed, with:
+
+```c
+NiFpga_Close(session, 0);
+```
+
+### 2) NiFpga_AcquireFifoReadElementsU16() function
+
+Here is an example to get elements from the `daqFIFO` using C/C++
+
+```c
+uint16_t* pData = NULL;
+size_t elementsRequested = 0;
+uint32_t timeout = 0;
+size_t elementsRemaining = 0;
+size_t elementsAcquired = 0;
+
+NiFpga_AcquireFifoReadElementsU16(session, NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO, &pData, elementsRequested, timeout, &elementsAcquired, &elementsRemaining);
+```
+where:
+- `session` handles a currently FPGA session
+- `NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO` is the FIFO from which to acquire
+- `&pData` outputs a pointer to the elements acquired
+- `elementsRequested` is the requested number of elements
+- `timeout` specifies the time in milliseconds the function waits for available data in the FIFO, if the FIFO is empty
+- `elementsAcquired` is the actual number of elements acquired
+- `elementsRemaining` outputs the number of elements remaining in the host memory part of FIFO
+
+Note that by using this function the user does not allocate memory of data.
+Instead, the system will transfer the memory to the user.
+Therefore, he/she needs to release acquired elements from FIFO.
+
+## How to release elements from FIFO
+
+The C API provides one function to release elements from FIFO [NiFpga_ReleaseFifoElements()](https://www.ni.com/docs/en-US/bundle/fpga-interface-c-api-ref/page/capi/functions_fifo_method.html).
+Only need this if the user uses the NiFpga_AcquireFifoReadElementsU16() function above.
+
+The NiFpga_AcquireFifoReadElementsU16() function will transfer FIFO’s memory to the user, and it is needed to release the elements from this memory region because it belongs to FIFO.
+The NiFpga_ReleaseFifoElements() function “releases previously acquired FIFO elements” according to the NiFpga_ReleaseFifoElements() [function description](https://www.ni.com/docs/en-US/bundle/fpga-interface-c-api-ref/page/capi/functions_fifo_method.html), and it says that “the host must release elements after acquiring them”.
+
+You are not supposed to use the NiFpga_ReleaseFifoElements()] function after the NiFpga_ReadFifoU16() function, because the NiFpga_ReadFifoU16() function allocates memory region by the user, therefore, the elements will not be in the FIFO’s memory when the NiFpga_ReleaseFifoElements() function tries to release those elements.
```

Comment:
Remove this sentence. It is not necessary to connect this concept to `read()` function. In the above, it is very clear to say only the `acquire()` function needs this.